pageserver/src/http/routes.rs

```diff
@@ -495,13 +495,39 @@ async fn get_lsn_by_timestamp_handler(
     let timeline = active_timeline_of_active_tenant(tenant_id, timeline_id).await?;
     let result = timeline.find_lsn_for_timestamp(timestamp_pg, &ctx).await?;
 
-    let result = match result {
-        LsnForTimestamp::Present(lsn) => format!("{lsn}"),
-        LsnForTimestamp::Future(_lsn) => "future".into(),
-        LsnForTimestamp::Past(_lsn) => "past".into(),
-        LsnForTimestamp::NoData(_lsn) => "nodata".into(),
-    };
-    json_response(StatusCode::OK, result)
+    if request
+        .headers()
+        .get(header::ACCEPT)
+        .map(|v| v == "application/json")
```

Comment:
I think new structure should be explained in the OpenAPI spec. It's possible to specify different return types there https://swagger.io/docs/specification/describing-responses/

Reply:
https://github.com/neondatabase/cloud/pull/7553